app/grandchallenge/workstations/templatetags/workstations.py

```diff
@@ -68,5 +71,11 @@ def workstation_query(
                 settings.WORKSTATIONS_CONFIG_QUERY_PARAM: reader_study.workstation_config.pk
             }
         )
+    elif display_set and display_set.reader_study.workstation_config:
+        query.update(
+            {
+                settings.WORKSTATIONS_CONFIG_QUERY_PARAM: display_set.reader_study.workstation_config.pk
+            }
+        )
```

Comment:
These are in the wrong place:

```
    elif reader_study and reader_study.workstation_config:
        query.update(
            {
                settings.WORKSTATIONS_CONFIG_QUERY_PARAM: reader_study.workstation_config.pk
            }
        )
    elif display_set and display_set.reader_study.workstation_config:
        query.update(
            {
                settings.WORKSTATIONS_CONFIG_QUERY_PARAM: display_set.reader_study.workstation_config.pk
            }
        )
```

If a config is explicitly set then it should take precedence over the defaults set by the reader study or display set.

Reply:
This should have a test! Maybe time to break up the mega 150 line test into digestible parts.

Reply:
But that is how it works now, explicitly defined configs take precedence over the defaults and there is a test for that as well. I will split the test up to make it easier to read.

Reply:
Ah yes I see, my mistake!

Reply:
It might be worth splitting this up into several template tags btw, given the rules here are so complex, so one for each of the entrypoints:

- image, with optional overlay
- reader_study, with optional user
- display_set
- algorithm_job
- archive_item

Each of them would allow an optional `config`